### PR TITLE
Issue #2513 - check bundle-level constraints on creates/updates

### PR DESCRIPTION
**Pull request by michaelwschroeder**
Signed-off-by: Mike Schroeder <mschroed@us.ibm.com>

### CODE REVIEW

File: fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java [hunk 1]
```diff
@@ -1570,8 +1582,42 @@ public FHIRPersistenceTransaction getTransaction() throws Exception {
...
+                    // Verify that the response field is not present for 'batch' or 'transaction'
+                    // type bundles (Bundle constraint bdl-4).
+                    if (requestEntry.getResponse() != null) {
+                        String msg = "Bundle.Entry.response must be empty.";
+                        throw buildRestException(msg, IssueType.INVALID);
+                    }
+
+                    // Verify that the fullUrl field is not a version specific reference
+                    // (Bundle constraint bdl-8) and that the fullUrl field + resource.meta.versionId
+                    // is unique for 'batch' or 'transaction' type bundles (Bundle constraint bdl-7)                    Resource resource = requestEntry.getResource();
```
Comment (lmsurpre) [inline fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java:1601]:
```suggestion
                    // is unique for 'batch' or 'transaction' type bundles (Bundle constraint bdl-7)
```

File: fhir-server/src/test/java/com/ibm/fhir/server/test/BundleValidationTest.java [hunk 2]
```diff
@@ -0,0 +1,692 @@
...
+
+public class BundleValidationTest {
+
+    FHIRPersistence persistence;
+    FHIRRestHelper helper;
+
+    @BeforeClass
+    void setup() throws FHIRException {
+        FHIRConfiguration.setConfigHome("src/test/resources");
+        FHIRRequestContext.get().setTenantId("default");
```
Comment (lmsurpre) [inline fhir-server/src/test/java/com/ibm/fhir/server/test/BundleValidationTest.java:55]: is this needed for the tests?

File: fhir-server/src/test/java/com/ibm/fhir/server/test/BundleValidationTest.java [hunk 3]
```diff
@@ -0,0 +1,692 @@
...
+
+        // Process request
+        FHIRRequestContext.get().setOriginalRequestUri("test");
+        FHIRRequestContext.get().setReturnPreference(HTTPReturnPreference.OPERATION_OUTCOME);
+        try {
+            helper.doBundle(requestBundle, false);
+            fail();
+        } catch (FHIROperationException e) {
+            // Validate results
+            // Validate results
```
Comment (lmsurpre) [inline fhir-server/src/test/java/com/ibm/fhir/server/test/BundleValidationTest.java:622]:
```suggestion
```

**Review by lmsurpre — APPROVED** [review]

**Review by prb112 — APPROVED** [review]
LGTM

File: fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java (hunk 1 above)
Comment (michaelwschroeder) [reply fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java]: done

File: fhir-server/src/test/java/com/ibm/fhir/server/test/BundleValidationTest.java (hunk 3 above)
Comment (michaelwschroeder) [reply fhir-server/src/test/java/com/ibm/fhir/server/test/BundleValidationTest.java]: done

File: fhir-server/src/test/java/com/ibm/fhir/server/test/BundleValidationTest.java (hunk 2 above)
Comment (michaelwschroeder) [reply fhir-server/src/test/java/com/ibm/fhir/server/test/BundleValidationTest.java]: not needed - removed unnecessary setup and teardown code

**Review by lmsurpre — APPROVED** [review]
LGTM

**Pull request merged**